pkg/types/types.go

```diff
@@ -40,3 +40,18 @@ type TransactionWithBytes struct {
 	T proto.Transaction
 	B []byte
 }
+
+// state for smart contracts
+type SmartState interface {
+	NewLastHeight() (uint64, error)
+	TransactionByID([]byte) (proto.Transaction, error)
+	TransactionHeightByID([]byte) (uint64, error)
+	Account(proto.Recipient) Account
+}
+
+// Some abstract way to get account related info
+type Account interface {
+	Data() []proto.DataEntry
```

Comment:
It isn't needed anywhere to retrieve all data entries at once. It can be heavy data.
Instead, we already have the following methods (`state/api.go`):
 ```
        // Accounts data storage.
        RetrieveNewestEntry(addr proto.Address, key string) (proto.DataEntry, error)
        RetrieveNewestIntegerEntry(addr proto.Address, key string) (*proto.IntegerDataEntry, error)
        RetrieveNewestBooleanEntry(addr proto.Address, key string) (*proto.BooleanDataEntry, error)
        RetrieveNewestStringEntry(addr proto.Address, key string) (*proto.StringDataEntry, error)
        RetrieveNewestBinaryEntry(addr proto.Address, key string) (*proto.BinaryDataEntry, error)
```